### PR TITLE
feat(controller): add pprof

### PR DESCRIPTION
**Pull request by yaroslavborbat**
## Description
add pprof

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: templates/virtualization-controller/_helpers.tpl [hunk 1]
```diff
@@ -36,4 +36,11 @@
...
 - name: UPLOADER_INGRESS_CLASS
   value: {{ include "helm_lib_module_ingress_class" . | quote }}
+{{/* {{- if eq .Values.virtualization.logLevel "debug" }}*/}}
+{{/*- name: PPROF_BIND_ADDRESS*/}}
+{{/*  value: ":8081"*/}}
+{{/* {{- else }}*/}}
+{{/* - name: PPROF_BIND_ADDRESS*/}}
+{{/*  value: ":8081"*/}}
+{{/* {{- end }}*/}}
 {{- end }}
```
Comment (diafour) [inline templates/virtualization-controller/_helpers.tpl:46]: Is it needed or can be fixed?

File: images/virtualization-artifact/hack/pyroscope.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+NAMESPACE=$(parse_flag "namespace" "n")
+SERVICE=$(parse_flag "service" "s")
+PORT=$(parse_flag "port" "p")
+
+cmd="docker compose"
+which docker-compose &>/dev/null && cmd="docker-compose"
+
+CMD="${ARGS[0]}"
+case "$CMD" in
```
Comment (diafour) [inline images/virtualization-artifact/hack/pyroscope.sh:84]:
```suggestion
case "${1}" in
```

File: images/virtualization-artifact/hack/pyroscope.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+if [[ $(parse_flag "help" "h") == "TRUE" ]]; then
+    usage_exit
+fi
+
+NAMESPACE=$(parse_flag "namespace" "n")
+SERVICE=$(parse_flag "service" "s")
+PORT=$(parse_flag "port" "p")
+
+cmd="docker compose"
```
Comment (diafour) [inline images/virtualization-artifact/hack/pyroscope.sh:80]: This cmd abstraction makes script more complicated. There are only 2 commands, you can simply use "docker compose" as-is without `+=` and `eval`:

```
function run() {
...
  docker compose -f ${DOCKER_COMPOSE_FILE} up -d
}
```

File: images/virtualization-artifact/hack/pyroscope.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+which docker-compose &>/dev/null && cmd="docker-compose"
```
Comment (diafour) [inline images/virtualization-artifact/hack/pyroscope.sh:81]: Why check for executable if you can run target command?

```suggestion
docker compose version &>/dev/null || {echo "No docker compose found" ; exit 1 }
```

File: images/virtualization-artifact/hack/pyroscope.sh (hunk 2 above)
Comment (yaroslavborbat) [reply images/virtualization-artifact/hack/pyroscope.sh:79]: We use args.sh to parse arguments. That's why we can run our scripts as pyroscope.sh --flag1=value1 run -- flag2=value2

**Review by diafour — APPROVED** [review]

**Review by diafour — APPROVED** [review]

**Pull request merged**